sdk/core/azure-core/src/main/java/com/azure/android/core/util/paging/PagedDataRetriever.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.util.paging;
+
+import java.util.Objects;
+
+/**
+ * Represents a synchronous paged data provider with fixed page size
+ * @param <T> items in the page
+ * @param <P> page of items
+ */
+public abstract class PagedDataRetriever<T, P extends Page<T>> {
+    private final int pageSize;
+
+    protected PagedDataRetriever(int pageSize) {
```

Comment:
We can probably use annotations such as `@NonNull` for this in addition to checking for nullness.

Reply:
This applies to all `PagedRetriever`s in this PR.